### PR TITLE
Bump nodejs-theforeman-vendor to 3.0.0

### PR DESCRIPTION
**Pull request by sharvit**
Needed for
- https://github.com/theforeman/foreman/pull/6981
- https://github.com/theforeman/foreman/pull/7164
- https://github.com/Katello/katello/pull/8430
- https://github.com/theforeman/foreman-tasks/pull/474
- https://github.com/theforeman/foreman_ansible/pull/311
- https://github.com/theforeman/foreman_remote_execution/pull/446
- https://github.com/theforeman/foreman_templates/pull/125

### CODE REVIEW

**Comment by sharvit** [comment]
Any idea what if wrong?

File: packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec [hunk 1]
```diff
@@ -1,24 +1,16 @@
-%{?scl:%scl_package nodejs-%{npm_name}}
-%{!?scl:%global pkg_name %{name}}
```
Comment (xprazak2) [inline packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec]: You need to keep these 2 lines.

File: packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec [hunk 2]
```diff
@@ -1,24 +1,16 @@
...
-
 %global npm_name @theforeman/vendor
 
-Name: %{?scl_prefix}nodejs-theforeman-vendor
```
Comment (xprazak2) [inline packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec]: keep the macro for scl prefix

File: packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec [hunk 3]
```diff
@@ -1,24 +1,16 @@
...
 Summary: foreman supported 3rd-party node_modules
 License: MIT
 Group: Development/Libraries
 URL: https://github.com/theforeman/foreman-js#readme
 Source0: https://registry.npmjs.org/%{npm_name}/-/%{npm_name}-%{version}.tgz
-%if 0%{?scl:1}
-BuildRequires: %{?scl_prefix_nodejs}npm
-%else
 BuildRequires: nodejs-packaging
-%endif
```
Comment (xprazak2) [inline packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec]: keep the whole if block

File: packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec [hunk 4]
```diff
@@ -1,24 +1,16 @@
...
 BuildArch: noarch
 ExclusiveArch: %{nodejs_arches} noarch
-Provides: %{?scl_prefix}npm(%{npm_name}) = %{version}
```
Comment (xprazak2) [inline packages/foreman/nodejs-theforeman-vendor/nodejs-theforeman-vendor.spec]: keep this line as well

**Comment by xprazak2** [comment]
Node packages were moved to scl, but the newer version of npm2rpm that reflects this change has not been released, so it updates your spec for the non-scl version.

**Comment by ekohl** [comment]
So we've updated [npm2rpm](https://github.com/theforeman/npm2rpm) to output specs that can work with SCL in order to support a more modern nodejs but currently rely on an unreleased version. If you generate this with the master version, it should work.

**Comment by sharvit** [comment]
Thanks @ekohl @xprazak2 

I followed @xprazak2 instructions

**Comment by sharvit** [comment]
It's green :)

**Comment by tbrisker** [comment]
should this be closed in favor of #4348 or update to get 3.1?

**Comment by sharvit** [comment]
Yes @tbrisker thanks, closing.

**Pull request closed without merging**